File: greencart_backend/src/modules/products/product.ts

import { Schema, Types, model } from 'mongoose';
export enum Category{
    Foodstuff="foodstuff",
    Electronics="electronics",
    Household="household",
    PersonalCare="personal-care",
}

export enum Metric{
    Kilograms="Kilograms",
    Centiliters="Centiliters",
    Mililiters="Mililiters",
    Grams="Grams",
}

export enum Rating{
    High="High",
    Medium="Medium",
    Low="Low"
}

export interface IProduct {
    _id: Types.ObjectId
    title: string
    type: string;
    thumbnail: string;
    carbon_emission: number
    category:Category
    price: number;

    metric:Metric
    quantity:number
    rating:Rating
}

const productSchema = new Schema<IProduct>({
    title: { type: String, required: true },
    type: { type: String, required: true },
    category: { type: String, required: true,enum:Category },
    carbon_emission: { type: Number, required: true },
    thumbnail: { type: String, required: true },
    price: { type: Number, required: true },
    metric: { type: String,enum:Metric },
    rating: { type: String,enum:Rating, required: true  },
    quantity: { type: Number},
});



export const Product = model<IProduct>('Product', productSchema);

File: greencart_mobile/lib/src/features/list_products/domain/models/product.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'product.freezed.dart';
part 'product.g.dart';

@freezed
class Product with _$Product {
  factory Product({
    @JsonKey(name: '_id') String? id,
    String? title,
    String? type,
    String? category,
    @JsonKey(name: 'carbon_emission') int? carbonEmission,
    String? thumbnail,
    int? price,
    String? metric,
    String? rating,
    int? quantity,
    @JsonKey(name: '__v') int? v,
  }) = _Product;

  factory Product.fromJson(Map<String, dynamic> json) =>
      _$ProductFromJson(json);

  static final dummyData = Product();
}
